tests/common.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2014, Joyent, Inc.
+ */
+
+/*
+ * Common functions used by tests.
+ */
+
+var execFile = require('child_process').execFile;
+
+
+function testFindSmartosImage(t, callback) {
+    var args = ['list', '-H', '-j', '-o', 'uuid,tags', 'os=smartos'];
+    var idx;
+    var img;
+    var imgs = {};
+    var latest;
+    var smartosImageUUID;
+
+    execFile('/usr/sbin/imgadm', args, function _onImgadm(err, stdout) {
+        t.ifError(err, 'load images from imgadm');
+        if (err) {
+            callback(smartosImageUUID);
+            return;
+        }
+
+        imgs = JSON.parse(stdout);
```

Comment:
If `imgadm` can output invalid JSON (like an error message not formatted as JSON), we might want to try/catch that `JSON.parse` and call `callback` with a non-null `err` in that case, instead of throwing an error.